docs/apis/plugintypes/repository/index.mdx

```diff
@@ -0,0 +1,975 @@
+- FILE_CONTROLLED_LINK - the file remains in the external repository. By "uploading" it, ownership of the file (in the remote system) is changed so that the [system account in the external repository](https://docs.moodle.org//en/OAuth\_2\_services) becomes the new owner of the file. Later, if the file is accessed, the system account is responsible for granting access to users.
+
+<Tabs>
+  <TabItem value="options" label="Setting options example" default>
+
+> Choices offered resulting from the values FILE_INTERNAL, FILE_REFERENCE, and FILE_CONTROLLED LINK, respectively. Whether FILE_EXTERNAL is present is never reflected in this list.
+>
+> ![Supported returntypes options settings](./_files/options.png)
+
+  </TabItem>
+  <TabItem value="code" label="View code" default>
+
+```php
+function supported_returntypes() {
+    return FILE_INTERNAL | FILE_EXTERNAL | FILE_REFERENCE | FILE_CONTROLLED_LINK;
+}
+```
+
+  </TabItem>
+</Tabs>
+
+The return values influence the choices offered to a user when selecting a file in file picker. Consider the screenshot above, which is from the dialogue that appears directly after choosing (but before uploading) a file in mod_resource. The three options result from FILE_INTERNAL, FILE_REFERENCE, and FILE_CONTROLLED_LINK being present. FILE_REFERENCE corresponds to the "alias/shortcut" option.
+
+The option FILE_EXTERNAL is never reflected in the file picker for mod_resource, so its absence or presence in supported_returntypes() is never reflected here. However, FILE_EXTERNAL is the only return type supported by mod_url: For mod_url, file picker will only(!) list repositories that support FILE_EXTERNAL.
+
+This implies that a plugin that uses a file picker is able to narrow the set of supported return types. For example, assignsubmission_file disallows FILE_EXTERNAL and FILE_REFERENCE.
+
+In the end, which type is
+used by Moodle depends on the choices made by the end user (e.g. inserting a link, will result in 'FILE_EXTERNAL'-related functions being used, using a 'shortcut/alias' will result in the 'FILE_REFERENCE'-related functions being used).
+
+#### supported_filetypes()
+
+Optional method that returns the kind of files a repository can store. Returns '\*' for all file types (default implementation), or an array of types or groups (e.g. array('text/plain', 'image/gif', 'web_image')).
+
+<Tabs>
+  <TabItem value="all" label="All files" default>
+
+```php
+function supported_filetypes() {
+    return '*';
+}
+```
+
+  </TabItem>
+  <TabItem value="mimetypes" label="Filter mimetypes" default>
+
+```php
+function supported_filetypes() {
+    return ['image/gif', 'image/jpeg', 'image/png'];
+}
+```
+
+  </TabItem>
+  <TabItem value="groups" label="Filter file groups" default>
+
+```php
+function supported_filetypes() {
+    return ['web_image'];
```

Comment:
Recommend adding some comments to the examples.

Reply:
Also, it's worth noting that you can add return type declaratinos to the child class, that is:
```
function supported_filetypes(): array {
}
```

It may be best if we do so to encourage better type hinting.

Reply:
Done